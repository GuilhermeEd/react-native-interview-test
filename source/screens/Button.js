import React, { Component } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import styles from './styles';

export default class Button extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onPress,
      text,
    } = this.props;
  
    return (
      <TouchableOpacity onPress={onPress}><View style={styles.button}><Text>{text}</Text></View></TouchableOpacity>
    )
  }
}
