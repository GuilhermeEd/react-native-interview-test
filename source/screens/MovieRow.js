import React, { Component } from 'react';
import { Text, View } from 'react-native';
import ButtonConainer from './ButtonContainer';
import styles from './styles';
import Button from './Button';

export default class MovieRow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDetails: false,
      liked: false,
    };
    this.toggleDetails = this.toggleDetails.bind(this);
  }

  toggleDetails() {
    this.setState({
      showDetails: !this.state.showDetails,
    });
  }

  render() {
    const {
      movie,
      likeMovie,
    } = this.props;

    return (
      <View>
        <Text style={styles.movieTitle}>{movie.title}</Text>
        {movie.liked && <Text style={styles.centerText}>Liked!</Text>}

        <ButtonConainer>
          <Button onPress={() => likeMovie(movie._id.$oid)} text={movie.liked ? 'Unlike' : 'Like'}/>
          <Button onPress={this.toggleDetails} text={'View Details'}/>
        </ButtonConainer>

        <View className='details' style={[styles.movieRowDetails, this.state.showDetails && styles.movieRowDetailsShown]}>
          <Text>Tickets available: {movie.inventory}</Text>
          <Text>Ticket price: ${movie.price}</Text>
        </View>
      </View>
    )
  }
}