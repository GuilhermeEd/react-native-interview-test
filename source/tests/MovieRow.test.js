import React from 'react';
import renderer from 'react-test-renderer';
import MovieRow from '../screens/MovieRow';
import Button from '../screens/Button';

const movieMock = {
  "_id": {
    "$oid": "5b8701a1fc13ae6569000000"
  },
  "title": "Long Live Death (Viva la muerte)",
  "price": 28.704,
  "inventory": 4,
};

const likeMovie = jest.fn();

test('<MovieRow> renders correctly', () => {
  const tree = renderer.create(<MovieRow movie={movieMock} likeMovie={likeMovie} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<MovieRow> like function gets called', () => {
  const rendered = renderer.create(<MovieRow movie={movieMock} likeMovie={likeMovie} />);
  const instance = rendered.root;
  const likeButton = instance.findAllByType(Button)[0];
  likeButton.props.onPress();
  expect(likeMovie.mock.calls.length).toBe(1);
});