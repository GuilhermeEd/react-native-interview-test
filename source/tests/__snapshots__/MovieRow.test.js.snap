// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieRow> renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    Long Live Death (Viva la muerte)
  </Text>
  <View
    className="buttonContainer"
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#cbe6e6",
            "borderRadius": 5,
            "flex": 1,
            "justifyContent": "center",
            "margin": 10,
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Like
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#cbe6e6",
            "borderRadius": 5,
            "flex": 1,
            "justifyContent": "center",
            "margin": 10,
            "padding": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          View Details
        </Text>
      </View>
    </View>
  </View>
  <View
    className="details"
    style={
      Array [
        Object {
          "height": 0,
          "overflow": "hidden",
          "paddingLeft": 10,
        },
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Tickets available: 
      4
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Ticket price: $
      28.704
    </Text>
  </View>
</View>
`;
